thoth-tech/doubtfire-web
# Migrate/unit ilo edit modal

# Description

This task migrates the `unit-ilo-edit-modal` component from CoffeeScript to TypeScript. This modal is used to add or edit learning outcomes that correspond to each unit, using `OutcomeLearningService` to associate it with a certain unit ID. The modal displays the title of the ILO, the abbreviation, and its description.

## Type of change
- [x] Frontend Migration


# How Has This Been Tested?
- Log in as `atutor` and select any unit.
- Go to the administration section of the unit, and then select the Learning Outcomes subtab.
- Either press the Add Outcome button or any pre-existing ILO. The former will create a blank ILO; the latter should display the ILO data.
- Click on save and the Learning Outcomes list should update itself.

## Previous modal display (edit mode)
![Initial ILO pre-migration](https://github.com/user-attachments/assets/d842cabf-53ec-417c-8dff-b599542e4648)

## Post-migration modal display (new, edit, and results)
![post-migration modal new](https://github.com/user-attachments/assets/e6c2bfcd-bd98-4ba1-9cbb-e6a594d589ef)
![post-migration modal edit](https://github.com/user-attachments/assets/d0e03848-efc9-4f3d-81f2-0198bbb8ac2a)
![post-migration modal result](https://github.com/user-attachments/assets/96a43aba-92c8-4193-b3cb-f46cb4056572)


## Testing Checklist:


# Checklist:
